drivers/storage/ec2/ec2.go

```diff
@@ -854,9 +872,20 @@ func (d *driver) CopySnapshot(runAsync bool,
 	}
 	resp := &ec2.CopySnapshotResp{}
 
-	auth := aws.Auth{
-		AccessKey: d.accessKey(),
-		SecretKey: d.secretKey()}
+	authFields := authEff(
+		map[string]interface{}{"moduleName": d.r.Context},
+		d.accessKey(),
+		d.secretKey(),
+	)
+	auth, err := aws.GetAuth(
+		d.accessKey(),
+		d.secretKey(),
+		"",
+		time.Now(),
+	)
+	if err != nil {
+		return nil, goof.WithFields(authFields, fmt.Errorf("error getting AWS credentials: %v", err).Error())
```

Comment:
Hi @branden,

Please avoid using `fmt.Errorf` and instead simply use Goof's fields and inner error function. Ex.

```
return nil, goof.WithFieldsE(authFields, "error getting AWS creds", err)
```